gravitee-am-common/src/main/java/io/gravitee/am/common/scim/parser/SCIMFilterParser.java

```diff
@@ -550,23 +546,20 @@ private static Object stringToValue(String string) {
         } else {
             char initial = string.charAt(0);
             if (initial >= '0' && initial <= '9' || initial == '-') {
-                try {
-                    if (isDecimalNotation(string)) {
-                        Double d = Double.valueOf(string);
-                        if (!d.isInfinite() && !d.isNaN()) {
-                            return d;
-                        }
-                    } else {
-                        Long myLong = Long.valueOf(string);
-                        if (string.equals(myLong.toString())) {
-                            if (myLong == (long)myLong.intValue()) {
-                                return myLong.intValue();
-                            }
-
-                            return myLong;
+                if (isDecimalNotation(string)) {
+                    Double d = Double.valueOf(string);
+                    if (!d.isInfinite() && !d.isNaN()) {
+                        return d;
+                    }
+                } else {
+                    Long myLong = Long.valueOf(string);
+                    if (string.equals(myLong.toString())) {
+                        if (myLong == (long) myLong.intValue()) {
+                            return myLong.intValue();
                         }
+
+                        return myLong;
                     }
-                } catch (Exception var3) {
```

Comment:
are you sure try catch can be removed? May it be better to add log in catch instead?